pkg/batch/subscriber/processor/segment_persister.go

```diff
@@ -0,0 +1,390 @@
+// Copyright 2024 The Bucketeer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package processor
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+
+	btclient "github.com/bucketeer-io/bucketeer/pkg/batch/client"
+	"github.com/bucketeer-io/bucketeer/pkg/batch/subscriber"
+	"github.com/bucketeer-io/bucketeer/pkg/feature/command"
+	"github.com/bucketeer-io/bucketeer/pkg/feature/domain"
+	v2fs "github.com/bucketeer-io/bucketeer/pkg/feature/storage/v2"
+	"github.com/bucketeer-io/bucketeer/pkg/pubsub"
+	"github.com/bucketeer-io/bucketeer/pkg/pubsub/publisher"
+	"github.com/bucketeer-io/bucketeer/pkg/pubsub/puller"
+	"github.com/bucketeer-io/bucketeer/pkg/pubsub/puller/codes"
+	"github.com/bucketeer-io/bucketeer/pkg/storage"
+	"github.com/bucketeer-io/bucketeer/pkg/storage/v2/mysql"
+	btproto "github.com/bucketeer-io/bucketeer/proto/batch"
+	domainproto "github.com/bucketeer-io/bucketeer/proto/event/domain"
+	serviceevent "github.com/bucketeer-io/bucketeer/proto/event/service"
+	featureproto "github.com/bucketeer-io/bucketeer/proto/feature"
+)
+
+const (
+	maxUserIDLength = 100
+)
+
+var (
+	errInvalidConfig           = errors.New("segment: invalid config")
+	errSegmentInUse            = errors.New("segment: segment is in use")
+	errExceededMaxUserIDLength = fmt.Errorf("segment: max user id length allowed is %d", maxUserIDLength)
+)
+
+type segmentPersisterConfig struct {
+	DomainEventProject string `json:"domainEventProject"`
+	DomainEventTopic   string `json:"domainEventTopic"`
+	FlushSize          int    `json:"flushSize"`
+	FlushInterval      int    `json:"flushInterval"`
+}
+
+type segmentPersister struct {
+	segmentPersisterConfig segmentPersisterConfig
+	domainPublisher        publisher.Publisher
+	batchClient            btclient.Client
+	mysqlClient            mysql.Client
+	logger                 *zap.Logger
+}
+
+func NewSegmentPersister(
```

Comment:
Please rename it as follows.
Same for the other places.

```suggestion
func NewSegmentUserPersister(
```

Reply:
Done!